build_library/vm_image_util.sh

```diff
@@ -579,6 +579,11 @@ install_oem_sysext() {
     for to_move in "${built_sysext_dir}/${oem_sysext}"*; do
         mv "${to_move}" "${upload_dir}/${to_move##*/}"
     done
+    # Generate dev-key-signed update payload for testing
+    delta_generator \
+      -private_key "/usr/share/update_engine/update-payload-key.key.pem" \
+      -new_image "${upload_dir}/${built_sysext_filename}" \
+      -out_file "${upload_dir}/flatcar_test_update-${oem_sysext}.gz"
```

Comment:
One question that comes to my mind is how this delta_generator figures out what to base the delta on? There's `-new_image`, but for deltas (diffs) you usually need two images, right?

Reply:
delta_generator has 2 different modes of generation, full update and partial delta update.

If an option `-old_image` is given, a partial delta update as you describe. If not given, it means implicitly a full update, so falls into the following code block.
https://github.com/flatcar/update_engine/blob/c6f566d47d8949632f7f43871eb8d5c625af3209/src/update_engine/delta_diff_generator.cc#L1436
I know, the code is so tricky. ;-)

Reply:
Thanks!